### ISSUE TITLE
Advanced simplification

### ISSUE THREAD

**Issue by sharkdp** [sharkdp/purescript-quantities#27]
For example:
```
Pa · m² -> N
J/s -> W
```

Idea: only apply simplifications that minimize the sum of the absolute value of all unit exponents.

**Comment by ghost** [comment]
@sharkdp  If I understood your conversation on reddit with doodle77 correctly: How about an explicit, unary simplification operator with a similar operator precedence like parentheses?

**Comment by sharkdp** [comment]
@sauer2 Yes, that would also be an option! Instead of the unary operator, I was thinking of something like "convert to SI" or "convert to base" with a syntax similar to the normal conversion operator

  *`<expr>`* `-> SI`
  *`<expr>`* `-> base`

**Comment by nimashoghi** [comment]
@sharkdp How about a combination of automatic deduction as well as some sort of user-defined deduction guide?

For example, `kg*m*s^(-2)` gets converted to N by default, but we can also provide a deduction guide.

Below is the format I was thinking of:
```
<unit to look for> [explicit] -> <unit to convert to>
kg*m*s^(-2) -> default // resets to automatic conversion
kg*m*s^(-2) -> delete // removes any conversion (same as kg*m*s^(-2) -> kg*m*s^(-2))
N -> kg*m*s^(-2) // converts N and all units equivalent to N to kg*m*s^(-2)
N explicit -> kg*m*s^(-2) // only converts N to kg*m*s^(-2), but not all equivalents.
```

**Comment by ghost** [comment]
@nimashoghi Sounds cool, but you probably want to adress @sharkdp .
